### PR TITLE
104-Added /auth/forgot/ tests

### PR DESCRIPTION
**Pull request by JPM1118**
#104 
_REQUIRES #84 to be merged_

- Tested for `200` status on non-existent user.
- Tested for `200` on existent user.
- Tested for token and token expiration creation for the user.

### CODE REVIEW

**Comment by donthedeveloper** [comment]
Check out these 2 PRs. Ben (mentor) left some fantastic feedback for Ben (other Ben) on writing and organizing tests.

https://github.com/donthedeveloper/site-builder/pull/83
https://github.com/donthedeveloper/site-builder/pull/80

Definitely check out his comments on how to organize these tests within describe blocks.

**Comment by JPM1118** [comment]
I read the comments from Ben's test PR's and I made some small changes, but I am still not entirely sure if I am writing these tests corrects. Hopefully, you can give me some feedback.

Also, should I be testing if node mailer sent a email successfully. If so, I not entirely sure how. Any pointers would be much appreciated. Thanks!

File: server/routes/api/__test__/auth.unit.test.js [hunk 1]
```diff
@@ -48,3 +48,34 @@ describe('POST Auth/Login', () => {
     expect(res.body.email).toEqual(email); // returned email should be correct user
   });
 });
+
+describe('POST Auth/Forgot', () => {
+  const existingUserEmail = 'testUser@test.com';
+
+  beforeAll(() => User.create({ email: existingUserEmail, password: 'test' }));
+
+  afterAll(() => User.findOneAndDelete({ email: existingUserEmail }));
```
Comment (donthedeveloper) [inline server/routes/api/__test__/auth.unit.test.js]: Minor formatting, but it usually is more readable with some spacing between these methods. Can you add a line break after this?

File: server/routes/api/__test__/auth.unit.test.js [hunk 2]
```diff
@@ -48,3 +48,34 @@ describe('POST Auth/Login', () => {
...
+  it('will return status 200 on nonexistent user', async () => {
+    const res = await request(app)
+      .post('/api/auth/forgot')
+      .type('form')
+      .send({ email: 'doesnotexist@test.com' });
+    expect(res.statusCode).toEqual(200);
+  }); // send 200 on non existent user email instead of error for security purposes.
```
Comment (donthedeveloper) [inline server/routes/api/__test__/auth.unit.test.js]: Usually a comment associated to a block of code will be above it. Let's place this comment above the 'it()'.

File: server/routes/api/__test__/auth.unit.test.js [hunk 3]
```diff
@@ -48,3 +48,34 @@ describe('POST Auth/Login', () => {
...
+  it('will return status 200 on existing user', async () => {
+    const res = await request(app)
+      .post('/api/auth/forgot')
+      .type('form')
+      .send({ email: existingUserEmail });
+    expect(res.statusCode).toEqual(200);
+  }); // send 200 on existent user email.
```
Comment (donthedeveloper) [inline server/routes/api/__test__/auth.unit.test.js]: This comment isn't needed. What's in the `it()` should describe this well enough.

File: server/routes/api/__test__/auth.unit.test.js [hunk 4]
```diff
@@ -48,3 +48,34 @@ describe('POST Auth/Login', () => {
...
+  it('will create and then save a token and expiration to user', async () => {
+    await request(app)
+      .post('/api/auth/forgot')
+      .type('form')
+      .send({ email: existingUserEmail });
+    const user = await User.findOne({ email: existingUserEmail }).exec();
+    expect(user.resetPassword.token && user.resetPassword.expiration).toBeTruthy();
+  }); /* token and expiration need to be assigned to
```
Comment (donthedeveloper) [inline server/routes/api/__test__/auth.unit.test.js]: Let's move this comment above as well.

**Pull request merged**